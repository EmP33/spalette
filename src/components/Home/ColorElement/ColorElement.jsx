import React, { useState } from "react";
import CSSModules from "react-css-modules";
import styles from "./ColorElement.module.scss";

const ColorElement = ({
  baseColor,
  testerColor,
  usedDivColor,
  colorName,
  usedInText,
}) => {
  const [text, setText] = useState("Copy");

  const copyToClipboardHandler = () => {
    navigator.clipboard.writeText(testerColor);
    setText(`Copied`);

    setTimeout(() => {
      setText("Copy");
    }, 2000);
  };
  return (
    <li>
      <div style={{ background: baseColor }}>
        {colorName}: {testerColor}
        <div
          onClick={copyToClipboardHandler}
          style={{ background: testerColor }}
          styleName="color-tester"
        >
          <h5>{text}</h5>
        </div>
      </div>
      <div style={{ background: usedDivColor, color: baseColor }}>
        {usedInText}
      </div>
    </li>
  );
};

export default CSSModules(ColorElement, styles);
