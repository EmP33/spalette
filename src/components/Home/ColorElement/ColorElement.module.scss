.color-tester {
  width: 15rem;
  height: 4rem;
  border: 3px solid #333;
  display: flex;
  align-items: center;
  justify-content: center;

  & > h5 {
    font-size: 1.4rem;
    color: #fff;
    background: #333;
    padding: 0.25rem;
    font-weight: 100;
    display: none;
  }

  &:hover {
    filter: brightness(125%);
    & > h5 {
      display: block;
    }
  }
}

.colors > li {
  width: 100%;
  padding: 2rem;
  display: flex;
  justify-content: space-between;

  & > div:nth-of-type(1) {
    width: 40%;
    padding: 1rem 2rem;
    font-size: 2rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;
  }
  & > div:nth-of-type(2) {
    width: 50%;
    padding: 1rem 2rem;
    font-size: 2rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
    cursor: pointer;
  }
}
