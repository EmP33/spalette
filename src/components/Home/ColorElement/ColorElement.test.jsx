import React from "react";
import { shallow } from "enzyme";
import ColorElement from "./ColorElement";

it("should render ColorElement component properly", () => {
  expect(
    shallow(
      <ColorElement
        baseColor="#111"
        testerColor="#111"
        usedDivColor="#111"
        colorName="Black"
        usedInText="Button"
      />
    )
  ).toMatchSnapshot();
});

Object.assign(navigator, {
  clipboard: {
    writeText: () => {},
  },
});

describe("clipboard", () => {
  beforeAll(() => {
    navigator.clipboard.writeText("#111");
  });
  it("should copy text to clipboard on click", () => {
    jest.spyOn(navigator.clipboard, "writeText");

    const wrapper = shallow(
      <ColorElement
        baseColor="#111"
        testerColor="#111"
        usedDivColor="#111"
        colorName="Black"
        usedInText="Button"
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.find(".color-tester").simulate("click");
    expect(navigator.clipboard.writeText).toHaveBeenCalledWith("#111");
  });
});
