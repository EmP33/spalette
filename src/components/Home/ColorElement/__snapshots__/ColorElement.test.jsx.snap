// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clipboard should copy text to clipboard on click 1`] = `
<li>
  <div
    style={
      Object {
        "background": "#111",
      }
    }
  >
    Black
    : 
    #111
    <div
      className="color-tester"
      onClick={[Function]}
      style={
        Object {
          "background": "#111",
        }
      }
    >
      <h5>
        Copy
      </h5>
    </div>
  </div>
  <div
    style={
      Object {
        "background": "#111",
        "color": "#111",
      }
    }
  >
    Button
  </div>
</li>
`;

exports[`should render ColorElement component properly 1`] = `
<li>
  <div
    style={
      Object {
        "background": "#111",
      }
    }
  >
    Black
    : 
    #111
    <div
      className="color-tester"
      onClick={[Function]}
      style={
        Object {
          "background": "#111",
        }
      }
    >
      <h5>
        Copy
      </h5>
    </div>
  </div>
  <div
    style={
      Object {
        "background": "#111",
        "color": "#111",
      }
    }
  >
    Button
  </div>
</li>
`;
